sysgen/sysgen.go

```diff
@@ -288,10 +288,13 @@ func generateArg(name, typ string, a []string, structs map[string]Struct, unname
 		}
 		fmt.Fprintf(out, "PtrType{%v, Dir: %v, Type: StrConstType{%v, Val: \"%v\"}}", common(), fmtDir("in"), common(), a[0]+"\\x00")
 	case "int8", "int16", "int32", "int64", "intptr":
-		if want := 0; len(a) != want {
-			failf("wrong number of arguments for %v arg %v, want %v, got %v", typ, name, want, len(a))
+		if len(a) == 2 {
```

Comment:
It is meant to be inclusive range, or exclusive range?
I think it's more intuitive to have an inclusive range.
Please extend description of int type on top of sys/sys.txt.

I agree that using flags for ranges is tedious.
But my concern is that once we give some meaning to arguments of a type, it will be difficult to change it (some users can have private syscall descriptions). E.g. if later we come up with better meaning for int arguments, we won't be able to change it.
I think that ultimately we want to support not only ranges, but arbitrary sets of values. For example:
```
foo int32[0, 5-10, 50, 100-200]
```
So please change the range format to `[begin-end]`. We will be able to extend such form the more general form.
You can use the following code for parsing:
https://play.golang.org/p/CQwd_G6keY